parlai/agents/transformer/modules/attention.py

```diff
@@ -14,10 +14,11 @@
 import torch.nn as nn
 import torch.nn.functional as F
 
+from parlai.agents.transformer.modules.interfaces import StaticComponent
 from parlai.utils.torch import neginf
 
 
-class BasicAttention(nn.Module):
+class BasicAttention(StaticComponent):
```

Comment:
I think this is the only thing I don't love of the latest version. are the StaticComponent annotations necessary?

Reply:
Not at all, it's just to differentiate between `ModularComponent`. Now that we've hidden the `ModularComponent` class, it probably makes sense to get rid of this.

cc @klshuster in case you feel strongly about keeping it

Reply:
don't feel strongly, feel free to get rid of it